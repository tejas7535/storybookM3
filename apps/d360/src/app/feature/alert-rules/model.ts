import { formatISO } from 'date-fns';

import { ResponseWithResultMessage } from '../../shared/utils/error-handling';

export interface AlertRuleResponse {
  count: number;
  content: AlertRule[];
}

export interface AlertRule {
  id: string;
  deleteData?: boolean | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  type?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  demandPlannerId?: string | null;
  demandPlanner?: string | null;
  gkamNumber?: string | null;
  gkamName?: string | null;
  startDate: Date | null;
  execDay?: ExecDay | null;
  execInterval?: ExecInterval | null;
  endDate: Date | null;
  deactivated?: boolean | null;
  activeCount?: number | null;
  inactiveCount?: number | null;
  completedCount?: number | null;
  alertComment?: string | null;
  region?: string | null;
  generation?: string | null;
  usernameCreated?: string | null;
  usernameLastChanged?: string | null;
  threshold1?: string | null;
  threshold1Description?: string | null;
  threshold1Type?: string | null;
  threshold2?: string | null;
  threshold2Description?: string | null;
  threshold2Type?: string | null;
  threshold3?: string | null;
  threshold3Description?: string | null;
  threshold3Type?: string | null;
  currency?: string | null;
}

export interface AlertTypeDescription {
  alertType: string;
  defaultName: string;
  description: string;
  generation: string;
  threshold1Description: string;
  threshold1Type: string;
  threshold2Description: string;
  threshold2Type: string;
  threshold3Description: string;
  threshold3Type: string;
}

export const execIntervallValues = [
  'M1',
  'M2',
  'M3',
  'M6',
  'W1',
  'D1',
] as const;
export type ExecInterval = (typeof execIntervallValues)[number];
export const execDayValues = ['M01', 'M15', 'W6', 'D'] as const;
export type ExecDay = (typeof execDayValues)[number];

export type AlertRuleSaveResponse = {
  id?: string | null;
  deleteData?: boolean | null;
  type?: string | null;
  region?: string | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  demandPlannerId?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  gkamNumber?: string | null;
  startDate: string | null;
  execDay?: string | null;
  execInterval?: string | null;
  endDate?: string | null;
  generation?: string | null;
  inactiveCount?: string | null;
  currency?: string | null;
  alertComment?: string | null;
  threshold1?: string | null;
  threshold2?: string | null;
  threshold3?: string | null;
  deactivated?: boolean | null;
} & ResponseWithResultMessage;

export interface AlertRuleSaveRequest {
  id: string;
  deleteData?: boolean | null;
  type?: string | null;
  region?: string | null;
  salesArea?: string | null;
  salesOrg?: string | null;
  customerNumber?: string | null;
  customerName?: string | null;
  materialNumber?: string | null;
  materialDescription?: string | null;
  materialClassification?: string | null;
  sectorManagement?: string | null;
  demandPlannerId?: string | null;
  productionLine?: string | null;
  productLine?: string | null;
  gkamNumber?: string | null;
  startDate?: string | null;
  execDay?: string | null;
  execInterval?: string | null;
  endDate?: string | null;
  generation?: string | null;
  inactiveCount?: string | null;
  currency?: string | null;
  alertComment?: string | null;
  threshold1?: string | null;
  threshold2?: string | null;
  threshold3?: string | null;
  deactivated?: boolean | null;
}

export function dataToAlertRuleRequest(data: AlertRule): AlertRuleSaveRequest {
  const startDate = data.startDate;
  const endDate = data.endDate;

  return {
    id: data.id,
    deleteData: data.deleteData,
    type: data.type,
    region: data.region,
    salesArea: data.salesArea,
    salesOrg: data.salesOrg,
    customerNumber: data.customerNumber,
    customerName: data.customerName,
    materialNumber: data.materialNumber,
    materialDescription: data.materialDescription,
    materialClassification: data.materialClassification,
    sectorManagement: data.sectorManagement,
    demandPlannerId: data.demandPlannerId,
    productionLine: data.productionLine,
    productLine: data.productLine,
    gkamNumber: data.gkamNumber,
    startDate: startDate
      ? formatISO(startDate, {
          representation: 'date',
        })
      : null,
    execDay: data.execDay,
    execInterval: data.execInterval,
    endDate: endDate
      ? formatISO(endDate, {
          representation: 'date',
        })
      : null,
    generation: data.generation,
    currency: data.currency,
    alertComment: data.alertComment,
    threshold1: data.threshold1,
    threshold2: data.threshold2,
    threshold3: data.threshold3,
    deactivated: data.deactivated,
  };
}
