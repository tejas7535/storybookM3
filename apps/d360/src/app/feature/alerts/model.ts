export interface Alert {
  id?: string;
  open: boolean;
  priority: boolean;
  alertPriority: Priority;
  deactivated: boolean;
  customerNumber?: string;
  customerName?: string;
  materialNumber?: string;
  materialDescription?: string;
  type?: AlertCategory;
  createdAt?: string;
  changedAt?: string;
  dueDate?: string;
  materialClassification?: string;
  customerMaterialNumber?: string;
  customerMaterialNumberCount?: number;
  openFunction?: OpenFunction;
  currency?: string;
  threshold1?: number;
  threshold1Description?: string;
  threshold2?: number;
  threshold2Description?: string;
  threshold3?: number;
  threshold3Description?: string;
  thresholdDeviation?: number;
  comment?: string;
  changedBy?: string;
  completedAt?: string;
  completedBy?: string;
  createdBy?: string;
  generation?: string;
  keyAccount?: string;
  regionPlPackage?: string;
  threshold1Type?: string;
  threshold2Type?: string;
  threshold3Type?: string;
  calcThreshold1?: number;
  calcThreshold2?: number;
  calcThreshold3?: number;
}

export enum AlertStatus {
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED',
  DEACTIVATED = 'DEACTIVATED',
}

export interface AlertNotificationCount {
  openCritical: number;
  openNonCritical: number;
}
export enum Priority {
  Priority1 = 1,
  Priority2 = 2,
  Priority3 = 3,
}

export enum OpenFunction {
  Validation_Of_Demand = 'VOD',
  Customer_Material_Portfolio = 'CMP',
}

export enum AlertCategory {
  CHKDMP = 'CHKDMP',
  DPLEOR = 'DPLEOR',
  DMPFIN = 'DMPFIN',
  DPOPDL = 'DPOPDL',
  CFPRAO = 'CFPRAO',
  CFSUAO = 'CFSUAO',
  PINWAO = 'PINWAO',
  PINWDP = 'PINWDP',
  PIOCAO = 'PIOCAO',
  PIOCDP = 'PIOCDP',
  PINUAO = 'PINUAO',
  PINUDP = 'PINUDP',
  PONWDP = 'PONWDP',
  IAUSDP = 'IAUSDP',
  IAUSAO = 'IAUSAO',
  ACIADP = 'ACIADP',
  SPNWAO = 'SPNWAO',
  SPSBDP = 'SPSBDP',
  REUDAO = 'REUDAO',
  REUSAO = 'REUSAO',
  RESEAO = 'RESEAO',
  RESEDP = 'RESEDP',
  SPCHDP = 'SPCHDP',
  SPCHAO = 'SPCHAO',
  DEVIFC = 'DEVIFC',
  SINWAO = 'SINWAO',
  NPOSDP = 'NPOSDP',
}

export const alertTypesToActivateToggleViaURL: AlertCategory[] = [
  AlertCategory.RESEDP, // Check substitution
  AlertCategory.REUSAO, // Check substitution, orders for predecessors
  AlertCategory.SINWAO, // Substitution Schaeffler intern activated
  AlertCategory.SPCHAO, // Substitution proposal changed
  AlertCategory.SPCHDP, // Substitution proposal changed
  AlertCategory.SPNWAO, // Check substitution proposal
  AlertCategory.SPSBDP, // Substitution proposal blocked
];
