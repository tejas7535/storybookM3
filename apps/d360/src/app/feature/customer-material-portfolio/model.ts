import { SortModelItem } from 'ag-grid-community';

import { ResponseWithResultMessage } from '../../shared/utils/error-handling';
import {
  DemandCharacteristic,
  MaterialClassification,
} from '../material-customer/model';
import { DemandPlanAdoption } from './cmp-modal-types';

/* Singe Phase In*/
export interface CMPSinglePhaseInRequest {
  customerNumber: string;
  materialNumber: string;
  /**
   * format is iso date string YYYY-MM-DDs
   */
  phaseInDate: string;
  demandCharacteristic: string;
}

export type CMPSinglePhaseInResponse = {
  customerNumber: string;
  materialNumber: string;
} & ResponseWithResultMessage;

/* Single Substitution*/
export type CMPSingleSubstitutionResponse = {
  customerNumber: string;
  materialNumber: string;
  confirmationNeeded: boolean;
} & ResponseWithResultMessage;

/* Bulk Phase In*/
export interface CMPBulkPhaseInRequest {
  customerNumber: string;
  phaseInEntities: CMPBulkPhaseInEntity[];
}

export interface CMPBulkPhaseInEntity {
  materialNumber: string;
  /**
   * format is iso date string YYYY-MM-DDs
   */
  phaseInDate: string;
  demandCharacteristic: string;
}

export interface CMPBulkPhaseInResponse {
  customerNumber: string;
  materialResults: CMPMaterialPhaseInResponse[];
}

export type CMPMaterialPhaseInResponse = {
  materialNumber: string;
} & ResponseWithResultMessage;

export interface CMPErrorMessage {
  materialNumber?: string;
  errorMessage: string;
}

/* Update CMPMaterial*/
export interface CMPWriteRequest {
  customerNumber: string;
  materialNumber: string;
  autoSwitchDate: string | null;
  repDate: string | null;
  portfolioStatus: string;
  demandCharacteristic: string | null;
  successorMaterial: string | null;
  demandPlanAdoption: DemandPlanAdoption | null;
}

export type CMPWriteResponse = {
  customerNumber: string;
  materialNumber: string;
} & ResponseWithResultMessage;

export interface CMPEntry {
  region?: string;
  salesArea?: string;
  sectorManagement?: string;
  salesOrg?: string;
  gkamNumber?: string;
  customerNumber?: string;
  materialNumber?: string;
  materialClassification?: MaterialClassification;
  sector?: string;
  productionSegment?: string;
  alertType?: string;
  materialDescription?: string;
  customerMaterialNumber?: string;
  customerMaterialNumberCount?: number;
  packagingSize?: number;
  productionPlant?: string;
  productionLine?: string;
  demandCharacteristic?: DemandCharacteristic;
  portfolioStatus?: string;
  pfStatusAutoSwitch?: string;
  repDate?: string;
  entryValidFrom?: string;
  stochasticType?: string;
  productLine?: string;
  productLineText?: string;
  deliveryQuantity18Months?: number;
  orderQuantity?: number;
  successorMaterial?: string;
  successorMaterialDescription?: string;
  successorCustomerMaterialNumber?: string;
  successorCustomerMaterialNumberCount?: number;
  successorMaterialPackagingSize?: number;
  successorMaterialClassification?: MaterialClassification;
  successorStochasticType?: string;
  successorProductionPlant?: string;
  successorProductionSegment?: string;
  successorProductionLine?: string;
  successorProductLine?: string;
  successorProductLineText?: string;
  successorSchaefflerMaterial?: string;
  successorSchaefflerMaterialDescription?: string;
  successorSchaefflerMaterialPackagingSize?: number;
  hasChildren?: boolean;
  tlMessageType: string | null;
  tlMessage: string | null;
  tlMessageNumber: number | null;
  tlMessageId: string | null;
  tlMessageV1: string | null;
  tlMessageV2: string | null;
  tlMessageV3: string | null;
  tlMessageV4: string | null;
}

export interface CfcrActionRequest {
  customerNumber: string;
  materialNumber: string;
  successorMaterial: string;
}

export interface CfcrActionResponse {
  cfcrActions: [
    {
      materialNumber: string;
      customerNumber: string;
      successorMaterial: string;
      cfcrAction: DemandPlanAdoption;
      selected: boolean;
    },
  ];
}

export interface CMPRequest {
  startRow: number | undefined;
  endRow: number | undefined;
  sortModel: SortModelItem[];
  selectionFilters: any;
  columnFilters: Record<string, any>;
}

export interface CMPResponse {
  headMaterials: { rows: CMPEntry[]; rowCount: number };
  childOfHeadMaterial: Map<string, CMPEntry[]>;
}
