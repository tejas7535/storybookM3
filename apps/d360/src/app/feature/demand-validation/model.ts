import { DateRange, DateRangePeriodType } from '../../shared/utils/date-range';
import { ResponseWithResultMessage } from '../../shared/utils/error-handling';
import { GlobalSelectionCriteriaFilters } from '../global-selection/model';
import { DemandValidationStringFilter } from './demand-validation-filters';

export type DemandValidationBatchResponse = {
  id: string;
  customerNumber: string;
  materialNumber: string;
  hasMultipleEntries: boolean;
  hasSuccessEntries: boolean;
  hasErrorEntries: boolean;
} & ResponseWithResultMessage;

export interface DemandValidationBatch {
  id: string;
  material: string;
  dateString: string;
  forecast: string;
  periodType: string;
  kpiEntries?: WriteKpiEntry[];
}

export interface BucketRequest {
  range1: KpiDataRequestDateRange;
  range2?: KpiDataRequestDateRange;
}

export interface KpiDataRequest {
  selectedKpis: SelectedKpis;
  customerNumber: string;
  materialNumber: string;
  range1: KpiDataRequestDateRange;
  range2?: KpiDataRequestDateRange;
  exceptions: string[];
}

export interface KpiDataRequestDateRange {
  from: string;
  to: string;
  period: DateRangePeriodType;
}

export type MaterialType = 'schaeffler' | 'customer';

export enum KpiBucketTypeEnum {
  WEEK = 'WEEK',
  MONTH = 'MONTH',
  PARTIAL_WEEK = 'PARTIAL_WEEK',
}

export type KpiBucketType =
  (typeof KpiBucketTypeEnum)[keyof typeof KpiBucketTypeEnum];

export interface MaterialListEntry {
  accountOwner?: string;
  alertType?: string;
  currency?: string;
  currentRLTSchaeffler?: string;
  customerClassification?: string;
  customerCountry?: string;
  customerMaterialNumber?: string;
  customerMaterialNumberCount?: number;
  customerName?: string;
  customerNumber?: string;
  dateBeginMaintPossibleMonth?: string;
  dateBeginMaintPossibleWeek?: string;
  dateEndMaintPossibleMonth?: string;
  dateEndMaintPossibleWeek?: string;
  dateFrozenZoneDl?: string;
  dateRltDl?: string;
  deliveryPlant?: string;
  demandCharacteristic?: string;
  demandPlanner?: string;
  demandPlanQuantity?: number;
  demandPlanQuantityUnit?: string;
  demandPlanValue?: number;
  dispoGroup?: string;
  fixHor?: string;
  frozenZone?: number;
  gkam?: string;
  gkamNumber?: string;
  gpsd?: string;
  gpsdName?: string;
  internalSales?: string;
  kam?: string;
  keyAccountName?: string;
  language?: string;
  mainCustomerName?: string;
  mainCustomerNumber?: string;
  materialClassification?: string;
  materialDescription?: string;
  materialNumber?: string;
  materialNumberS4?: string;
  packagingSize?: number;
  planningPlant?: string;
  portfolioStatus?: string;
  portfolioStatusDate?: string;
  productCluster?: string;
  productionLine?: string;
  productionPlant?: string;
  productionPlantName?: string;
  productionSegment?: string;
  productLine?: string;
  productLineName?: string;
  region?: string;
  safetyStock?: number;
  safetyStockCustomer?: number;
  salesArea?: string;
  salesOrg?: string;
  sector?: string;
  sectorDescription?: string;
  sectorManagement?: string;
  sectorManagementText?: string;
  stochasticType?: string;
  subKeyAccount?: string;
  subKeyAccountName?: string;
  successorCustomerMaterial?: string;
  successorCustomerMaterialDescription?: string;
  successorSchaefflerMaterial?: string;
  successorSchaefflerMaterialDescription?: string;
  transitTimeBetweenProdPlantAndDistributionPlant?: number;
}

export interface KpiEntry {
  bucketType: KpiBucketType;
  confirmedSalesPlan: number | null;
  confirmedDeliveriesActive: number | null;
  confirmedDeliveriesCombined: number | null;
  confirmedDeliveriesPredecessor: number | null;
  confirmedDemandPlan: number | null;
  confirmedDemandRelevantSales: number | null;
  confirmedFirmBusinessActive: number | null;
  confirmedFirmBusinessCombined: number | null;
  confirmedFirmBusinessPredecessor: number | null;
  confirmedOnTopCapacityForecast: number | null;
  confirmedOnTopOrder: number | null;
  confirmedOpportunities: number | null;
  confirmedSalesAmbition: number | null;
  currentDemandPlan: number | null;
  salesPlan: number | null;
  deliveriesActive: number | null;
  deliveriesCombined: number | null;
  deliveriesPredecessor: number | null;
  demandRelevantSales: number | null;
  firmBusinessActive: number | null;
  firmBusinessCombined: number | null;
  firmBusinessPredecessor: number | null;
  forecastProposal: number | null;
  forecastProposalDemandPlanner: number | null;
  fromDate: string;
  indicativeDemandPlan: number | null;
  onTopCapacityForecast: number | null;
  onTopOrder: number | null;
  opportunities: number | null;
  salesAmbition: number | null;
  storedBucketType: KpiBucketType | null;
  toDate: string;
  validatedForecast: number | null;
}

export interface KpiData {
  customerNumber: string;
  materialNumber: string;
  data: KpiEntry[];
}

export interface KpiDateRanges {
  range1: DateRange;
  range2?: DateRange;
}

export interface KpiBucket {
  from: string;
  to: string;
  type: KpiBucketType;
}

export interface WriteKpiData {
  ids?: string[];
  customerNumber: string;
  materialNumber: string;
  kpiEntries: WriteKpiEntry[];
}

export interface WriteKpiEntry {
  idx?: number;
  fromDate: string | null;
  bucketType: KpiBucketType;
  validatedForecast: number | null;
}

export interface WriteKpiDataResponse {
  ids?: string[];
  customerNumber: string;
  materialNumber: string;
  results: WriteKpiEntryResult[];
}

export type WriteKpiEntryResult = {
  idx?: number;
  fromDate: string;
} & ResponseWithResultMessage;

export enum KpiType {
  ActiveAndPredecessor = 'activeAndPredecessor',
  Deliveries = 'deliveries',
  FirmBusiness = 'firmBusiness',
  Opportunities = 'opportunities',
  ForecastProposal = 'forecastProposal',
  ForecastProposalDemandPlanner = 'forecastProposalDemandPlanner',
  ValidatedForecast = 'validatedForecast',
  DemandRelevantSales = 'demandRelevantSales',
  OnTopOrder = 'onTopOrder',
  OnTopCapacityForecast = 'onTopCapacityForecast',
  SalesAmbition = 'salesAmbition',
  SalesPlan = 'salesPlan',
  ConfirmedDeliveries = 'confirmedDeliveries',
  ConfirmedFirmBusiness = 'confirmedFirmBusiness',
  ConfirmedDemandRelevantSales = 'confirmedDemandRelevantSales',
  ConfirmedOnTopOrder = 'confirmedOnTopOrder',
  ConfirmedOnTopCapacityForecast = 'confirmedOnTopCapacityForecast',
  ConfirmedSalesAmbition = 'confirmedSalesAmbition',
  ConfirmedOpportunities = 'confirmedOpportunities',
  ConfirmedSalesPlan = 'confirmedSalesPlan',
}
export type SelectedKpis = Record<KpiType, boolean>;

export interface DeleteKpiDataRequest {
  customerNumber: string;
  fromDate: string;
  toDate: string;
  materialNumbers: string[];
}

export type DeleteKpiDataResponse = {
  customerNumber: string;
  fromDate: string;
  toDate: string;
  results: DeleteKpiEntryResult[];
} & ResponseWithResultMessage;

export type DeleteKpiEntryResult = {
  materialNumber: string;
} & ResponseWithResultMessage;

export interface DemandMaterialCustomerRequest {
  selectionFilters:
    | (GlobalSelectionCriteriaFilters & DemandValidationStringFilter)
    | undefined;
  startRow: number;
  endRow: number;
  sortModel: object[];
}

export const SUPPLY_CONCEPT_SUPPORTED_STOCHASTIC_TYPES = [
  'C',
  'E',
  'F',
  'S',
  'U',
  'M',
  'N',
] as const;

export type SupplyConceptsStochasticType =
  (typeof SUPPLY_CONCEPT_SUPPORTED_STOCHASTIC_TYPES)[number];
