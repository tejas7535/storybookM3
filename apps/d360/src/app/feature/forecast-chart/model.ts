import { SeriesOption } from 'echarts';

import {
  dimmedBlue,
  dimmedGreen,
  dimmedGrey,
  dimmedPurple,
  dimmedRed,
  dimmedYellow,
  textDarkGrey,
} from '../../shared/styles/colors';
import { PlanningView } from '../demand-validation/planning-view';

export interface ForecastChartData {
  chartUnitMode: ChartUnitMode;
  planningView: PlanningView;
  currency: string;
  chartEntries: MonthlyChartEntry[];
}

export const chartSeriesConfig = {
  deliveries: { color: dimmedGrey, isToggleable: false, order: 1 },
  orders: { color: dimmedYellow, isToggleable: false, order: 2 },
  onTopOrder: { color: dimmedGreen, isToggleable: true, order: 3 },
  onTopCapacityForecast: { color: dimmedBlue, isToggleable: true, order: 4 },
  salesAmbition: { color: dimmedPurple, isToggleable: true, order: 5 },
  opportunities: { color: dimmedRed, isToggleable: true, order: 6 },
  salesPlan: { color: textDarkGrey, isToggleable: false, order: 7 },
} as const;

export enum KpiValues {
  Deliveries = 'deliveries',
  Orders = 'orders',
  OnTopOrder = 'onTopOrder',
  OnTopCapacityForecast = 'onTopCapacityForecast',
  SalesAmbition = 'salesAmbition',
  Opportunities = 'opportunities',
  SalesPlan = 'salesPlan',
}

export type ChartValues = keyof typeof chartSeriesConfig;

export interface ChartEntry {
  orders: number;
  deliveries: number;
  onTopOrder: number;
  onTopCapacityForecast: number;
  opportunities: number;
  salesAmbition: number;
  salesPlan: number | null;
}

export type MonthlyChartEntry = ChartEntry & {
  yearMonth: string;
};

export type YearlyChartEntry = ChartEntry & {
  year: number;
};

export type KpiSeriesOption = SeriesOption & { kpi?: string };

export enum ChartUnitMode {
  CURRENCY = 'CURRENCY',
  QUANTITY = 'QUANTITY',
}

export enum PeriodType {
  YEARLY = 'YEARLY',
  MONTHLY = 'MONTHLY',
}

export interface ChartSettings {
  startDate: Date | string;
  endDate: Date | string;
  planningView: PlanningView;
  chartUnitMode: ChartUnitMode;
  periodType: PeriodType;
}

export interface ChartSettingsStored {
  startDate: string;
  endDate: string;
  planningView: string;
  chartUnitMode: string;
  periodType: string;
}
