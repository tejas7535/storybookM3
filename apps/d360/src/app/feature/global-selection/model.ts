import { GlobalSelectionStateService } from '../../shared/components/global-selection-criteria/global-selection-state.service';
import { SelectableValue } from '../../shared/components/inputs/autocomplete/selectable-values.utils';

export interface CustomerEntry {
  salesOrg?: string;
  customerNumber: string;
  customerName?: string;
  planningCurrency?: string;
}

export type GlobalSelectionCriteriaFilters = Partial<
  Record<(typeof GlobalSelectionStateService.stateKeys)[number], string[]>
>;

export type GlobalSelectionCriteriaFields = Partial<
  Record<
    (typeof GlobalSelectionStateService.stateKeys)[number],
    SelectableValue[]
  >
>;

export enum GlobalSelectionStatus {
  DATA_AVAILABLE = 'DATA_AVAILABLE',
  DATA_NO_RESULTS = 'DATA_NO_RESULTS',
  DATA_LOADING = 'DATA_LOADING',
  DATA_ERROR = 'DATA_ERROR',
  DATA_NOTHING_SELECTED = 'DATA_NOTHING_SELECTED',
}

export enum Region {
  Europe = 'EU',
  GreaterChina = 'GC',
  AsiaPacific = 'AP',
  Americas = 'AM',
}
