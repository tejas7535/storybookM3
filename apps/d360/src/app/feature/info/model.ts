// original file path: /src/domain/info/useCurrency.ts
// TODO align with RenÃ¨ if it is ok to delete all the types from this file
export interface CurrencyData {
  currentCurrency: string;
  availableCurrencies: string[];
  setCurrentCurrency: (currency: string) => void;
}
