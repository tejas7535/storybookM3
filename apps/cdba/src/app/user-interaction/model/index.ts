export * from './action.model';
export * from './interaction.model';
export * from './interaction-type.enum';
