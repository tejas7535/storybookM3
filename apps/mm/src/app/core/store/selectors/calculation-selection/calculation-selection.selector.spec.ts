import { Bearing } from '../../models/calculation-selection-state.model';
import {
  getBearing,
  getBearingSeatId,
  getBearingSeats,
  getBearingSelectionLoading,
  getBearingsResultList,
  getCurrentStep,
  getMeasurementMethod,
  getMeasurementMethods,
  getMountingMethod,
  getMountingMethods,
  getStepperState,
  getSteps,
} from './calculation-selection.selector';

describe('Calculation Selection Selectors', () => {
  const initialState = {
    stepper: { currentStep: 2 },
    bearing: { id: 'bearing1' },
    bearingResultList: [{ id: 'result1' }],
    loading: true,
    bearingSeats: { selectedValueId: 'seat1', list: [{ id: 'seat1' }] },
    measurementMethods: {
      selectedValueId: 'method1',
      list: [{ id: 'method1' }],
    },
    mountingMethods: {
      selectedValueId: 'mounting1',
      list: [{ id: 'mounting1' }],
    },
  };

  const state = {
    calculationSelection: initialState,
  };

  it('should select the stepper state', () => {
    const result = getStepperState(state);
    expect(result).toEqual(initialState.stepper);
  });

  it('should select the current step', () => {
    const result = getCurrentStep(state);
    expect(result).toEqual(initialState.stepper.currentStep);
  });

  it('should select the bearing', () => {
    const result = getBearing(state);
    expect(result).toEqual(initialState.bearing);
  });

  it('should select the bearings result list', () => {
    const result = getBearingsResultList(state);
    expect(result).toEqual(initialState.bearingResultList);
  });

  it('should select the loading state', () => {
    const result = getBearingSelectionLoading(state);
    expect(result).toEqual(initialState.loading);
  });

  it('should select the bearing seat id', () => {
    const result = getBearingSeatId(state);
    expect(result).toEqual(initialState.bearingSeats.selectedValueId);
  });

  it('should select the bearing seats', () => {
    const result = getBearingSeats(state);
    expect(result).toEqual(initialState.bearingSeats);
  });

  it('should select the measurement method id', () => {
    const result = getMeasurementMethod(state);
    expect(result).toEqual(initialState.measurementMethods.selectedValueId);
  });

  it('should select the measurement methods', () => {
    const result = getMeasurementMethods(state);
    expect(result).toEqual(initialState.measurementMethods);
  });

  it('should select the mounting methods', () => {
    const result = getMountingMethods(state);
    expect(result).toEqual(initialState.mountingMethods);
  });

  it('should select the mounting method id', () => {
    const result = getMountingMethod(state);
    expect(result).toEqual(initialState.mountingMethods.selectedValueId);
  });

  describe('getSteps Selector', () => {
    const initState = {
      bearing: { id: 'bearing1' } as Partial<Bearing> as Bearing,
      bearingSeatId: 'seat1',
      mountingMethod: 'method1',
      measurementMethod: 'LB_AXIAL_DISPLACEMENT',
      isAvailable: true,
    };

    it('should return steps with correct properties', () => {
      const result = getSteps.projector(
        initState.bearing,
        initState.bearingSeatId,
        initState.mountingMethod,
        initState.measurementMethod,
        initState.isAvailable
      );

      expect(result).toMatchSnapshot();
    });
  });
});
