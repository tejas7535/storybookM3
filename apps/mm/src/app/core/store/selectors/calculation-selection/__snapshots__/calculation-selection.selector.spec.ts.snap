// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation Selection Selectors getSteps Selector should return steps with correct properties 1`] = `
Array [
  Object {
    "complete": Object {
      "id": "bearing1",
    },
    "customContent": "bearing-selection",
    "editable": true,
    "enabled": true,
    "index": 0,
    "name": "RSY_PAGE_BEARING_TYPE",
    "text": "dialog.bearing",
  },
  Object {
    "complete": "seat1",
    "customContent": "bearing-seat",
    "editable": true,
    "enabled": true,
    "index": 1,
    "name": "PAGE_MOUNTING_MANAGER_SEAT",
    "text": "dialog.bearingSeat",
  },
  Object {
    "complete": "method1",
    "customContent": "measuring-and-mounting",
    "editable": true,
    "enabled": true,
    "index": 2,
    "name": "PAGE_MOUNTING_MANAGER_MEASURING_MOUTING_METHODS",
    "text": "dialog.measuringAndMounting",
  },
  Object {
    "complete": true,
    "customContent": "calculation-options",
    "editable": true,
    "enabled": true,
    "index": 3,
    "name": "PROPERTY_PAGE_MOUNTING_SITUATION",
    "text": "dialog.calculationOptions",
  },
  Object {
    "complete": false,
    "customContent": "calculation-result",
    "editable": true,
    "enabled": true,
    "index": 4,
    "name": "PAGE_RESULT",
    "text": "dialog.report",
  },
]
`;
